### PR TITLE
Support Ebpf Hash and Checksum

### PR DESCRIPTION
**Pull request by komaljai**
Generate checksum functions EBPF is using for checksum and hash in TC.
Introduce functions:

crc16_update
crc16_finalize
crc32_update
crc32_finalize
csum16_add
csum16_sub

### CODE REVIEW

**Review by psivanup — APPROVED** [review]
looks fine

File: backends/tc/ebpfCodeGen.cpp [hunk 1]
```diff
@@ -1529,6 +1560,41 @@ void ControlBodyTranslatorPNA::processApply(const P4::ApplyMethod *method) {
...
+        auto mi = P4::MethodInstance::resolve(methodCallExpr, control->program->refMap,
+                                              control->program->typeMap);
+        auto ext = mi->to<P4::ExternMethod>();
+        if (ext == nullptr) {
+            return false;
+        }
+
+        if (ext->originalExternType->name.name == "Hash") {
+            cstring name = EBPF::EBPFObject::externalName(ext->object);
+            auto hash = control->to<EBPF::EBPFControlPSA>()->getHash(name);
```
Comment (usha1830) [inline backends/tc/ebpfCodeGen.cpp:1598]: Is this supposed to be PSA or PNA?

**Review by usha1830 — APPROVED** [review]
Looks fine

File: backends/tc/ebpfCodeGen.cpp (hunk 1 above)
Comment (komaljai) [reply backends/tc/ebpfCodeGen.cpp:1598]: PSA

**Comment by komaljai** [comment]
Merged to main, Closing.

**Pull request closed without merging**